src/events/event_handler.ts

```diff
@@ -127,6 +128,59 @@ export class eventHandler {
 		this.handleChanges(filePath, currentText);
 	}
 
+	handleTabChangeEvent = () => {
+		/*
+		The flow for handler will be as follows:
+			- For the current open repoPath, get the repo_id and from config File
+			- Get Current Open Tabs TabGroups.ALL
+			- For each tab:
+				* Get the Relative Path of each file using repoPath
+				* Get the fileID using relative_path found in above step
+		*/
+		// Record timestamp
+		const created_at = new Date().getTime();
+		console.log("created_at: ", created_at);
+		// For the current open repoPath, get the repo_id and from config File
+		const configUtils = new ConfigUtils();
+		const repoId = configUtils.getRepoIdByPath(this.repoPath);
+		console.log(`Repo ID: ${repoId}`)
+		const configJSON = configUtils.config;
+
+		// Get list of current tabs
+		const open_tabs = vscode.window.tabGroups.all;
+		let tabs: any[] = [];
+		// Loop through tab groups
+		for (const tab_group of open_tabs) {
+			for (let tab of tab_group.tabs) {
+				// console.log(`Displaying tabs: `, tab);
+				// Get path of tab
+				// @ts-ignore
+				let fileRelativePath = this.pathUtils.getFileRelativePath(tab.input.uri.path);
+				// console.log("File Name: ", fileRelativePath);
+				// Get file ID using path
+				let fileId = configUtils.getFileIdByPath(this.repoPath, this.branch, fileRelativePath);
+				// console.log("File ID: ", fileId);
+				tabs.push({"file_id": fileId, "path": fileRelativePath});
+			}
+		}
+		const tabsYAML = yaml.dump(tabs);
```

Comment:
Do we need to convert to YML here? Maybe we create 1 big JSON and dump to YML only once

Reply:
Can you elaborate how we will create 1 JSON?

I tried with `JSON.stringify`, but the format is not coming as intended:

![image](https://github.com/codesyncapp/codesync-vs/assets/15974895/971a6af8-1d5f-471c-9050-2c1eff9cdc38)

Reply:
Idea is to have following json
{
    "repo_id": 1234
    "created_at": timestamp_str
    "source": vscode
    "tabs": [
        {
              "file_id": 1234
              "path": "abc/def"
         }
     ]
}